pkg/controller/istio/istio_controller.go

```diff
@@ -0,0 +1,465 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/stringsutil"
+	"github.com/go-logr/logr"
+	"github.com/tigera/api/pkg/lib/numorstring"
+	operatorv1 "github.com/tigera/operator/api/v1"
+	v1 "github.com/tigera/operator/api/v1"
+	crdv1 "github.com/tigera/operator/pkg/apis/crd.projectcalico.org/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/controller/utils/imageset"
+	"github.com/tigera/operator/pkg/ctrlruntime"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/gatewayapi"
+	renderistio "github.com/tigera/operator/pkg/render/istio"
+)
+
+const (
+	reasonInfo = "Info_reconciling_Tigera_Istio"
+	reasonErr  = "Error_reconciling_Tigera_Istio"
+)
+
+var (
+	// blank assignment to verify that ReconcileIstio implements reconcile.Reconciler
+	_ reconcile.Reconciler = &ReconcileIstio{}
+
+	log = logf.Log.WithName("controller_istio")
+)
+
+// Add creates a new Istio Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+//
+// Start Watches within the Add function for any resources that this controller creates or monitors. This will trigger
+// calls to Reconcile() when an instance of one of the watched resources is modified.
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	r := newReconciler(mgr, opts)
+
+	c, err := ctrlruntime.NewController("istio-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return fmt.Errorf("failed to create istio-controller: %w", err)
+	}
+
+	// Watch for changes to primary resource Istio
+	err = c.WatchObject(&operatorv1.Istio{}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		log.V(5).Info("Failed to create Istio watch", "err", err)
+		return fmt.Errorf("istio-controller failed to watch primary resource: %v", err)
+	}
+
+	if err = utils.AddInstallationWatch(c); err != nil {
+		log.V(5).Info("Failed to create network watch", "err", err)
+		return fmt.Errorf("istio-controller failed to watch Tigera network resource: %v", err)
```

Comment:
I think these are copy/paste issues, but this isn't called `Network` any more :) 
```suggestion
		return fmt.Errorf("istio-controller failed to watch Installation resource: %v", err)
```